src/Symfony/Bridge/Twig/Extension/WebLinkExtension.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Twig\Extension;
+
+use Fig\Link\GenericLinkProvider;
+use Fig\Link\Link;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\WebLink\WebLinkManagerInterface;
+
+/**
+ * Twig extension for the Symfony WebLink component.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+class WebLinkExtension extends \Twig_Extension
+{
+    private $requestStack;
+
+    public function __construct(RequestStack $requestStack)
+    {
+        $this->requestStack = $requestStack;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getFunctions()
+    {
+        return array(
+            new \Twig_SimpleFunction('link', array($this, 'link')),
+            new \Twig_SimpleFunction('preload', array($this, 'preload')),
+            new \Twig_SimpleFunction('dns_prefetch', array($this, 'dnsPrefetch')),
+            new \Twig_SimpleFunction('preconnect', array($this, 'preconnect')),
+            new \Twig_SimpleFunction('prefetch', array($this, 'prefetch')),
+            new \Twig_SimpleFunction('prerender', array($this, 'prerender')),
+        );
+    }
+
+    /**
+     * Adds a "Link" HTTP header.
+     *
+     * @param string $uri        The relation URI
+     * @param string $rel        The relation type (e.g. "preload", "prefetch", "prerender" or "dns-prefetch")
+     * @param array  $attributes The attributes of this link (e.g. "array('as' => true)", "array('pr' => 0.5)")
+     *
+     * @return string The relation URI
+     */
+    public function link($uri, $rel, array $attributes = array())
+    {
+        if (!$request = $this->requestStack->getMasterRequest()) {
+            return $uri;
+        }
+
+        $link = new Link($rel, $uri);
+        foreach ($attributes as $key => $value) {
+            $link = $link->withAttribute($key, $value);
+        }
+
+        $linkProvider = $request->attributes->get('_links', new GenericLinkProvider());
```

Comment:
Actually, I really think the ideal case would be for `Symfony\Component\HttpFoundation\Request` to implement `Psr\Link\EvolvableLinkProviderInterface`. That's what I understand after reading PSR-13.

But I understand that's not probably not possible considering the mutable nature of our `Request`. That said, I think using attributes for this is wrong.